### PR TITLE
add links to interesting items in `std::ptr` documentation

### PR DESCRIPTION
**Pull request by matthew-piziak**
Part of https://github.com/rust-lang/rust/issues/29371.

r? @steveklabnik

### CODE REVIEW

File: src/libcore/intrinsics.rs [hunk 1]
```diff
@@ -224,15 +224,17 @@ extern "rust-intrinsic" {
     ///   trait objects, because they can't be read out onto the stack and
     ///   dropped normally.
     ///
-    /// * It is friendlier to the optimizer to do this over `ptr::read` when
+    /// * It is friendlier to the optimizer to do this over [`ptr::read`] when
     ///   dropping manually allocated memory (e.g. when writing Box/Rc/Vec),
```
Comment (GuillaumeGomez) [inline src/libcore/intrinsics.rs]: Can you add urls for Box, Rc and Vec as well please? (If you really don't want to, I won't bother you with it.)

**Comment by GuillaumeGomez** [comment]
Broken:

```bash
std\ptr\fn.read.html:59: broken link - std\ptr\fn.zero_memory.html
std\ptr\fn.read.html:59: broken link - std\ptr\fn.copy_memory.html
std\ptr\fn.read_volatile.html:69: broken link - std\ptr\fn.zero_memory.html
std\ptr\fn.read_volatile.html:69: broken link - std\ptr\fn.copy_memory.html
```

(and some others)

**Comment by matthew-piziak** [comment]
@GuillaumeGomez What is the correct command to reproduce this breakage? I cannot reproduce with `make docs`.

**Comment by GuillaumeGomez** [comment]
```bash
cd src/tools/linkchecker
cargo run ../../lib_you_want_to_check
```

**Comment by matthew-piziak** [comment]
Have `copy_memory` and `zero_memory` been removed? (I wouldn't be surprised. They were experimental before.)

**Comment by GuillaumeGomez** [comment]
Check on nigthly docs.

**Comment by matthew-piziak** [comment]
@GuillaumeGomez I tried that. I'd expect to see them here: https://doc.rust-lang.org/nightly/std/mem/

**Comment by GuillaumeGomez** [comment]
These two functions have been removed it seems, indeed.

**Comment by steveklabnik** [comment]
Travis looks spurious, let's try with bors.

@bors: r+ rollup

**Comment by jonathandturner** [comment]
@bors r-

Looks like there some breakages on x86_64-unknown-linux-gnu

```
core/intrinsics/fn.drop_in_place.html:59: broken link - core/intrinsics/fn.read.html
core/intrinsics/fn.drop_in_place.html:61: broken link - core/boxed/struct.Box.html
core/intrinsics/fn.drop_in_place.html:61: broken link - core/rc/struct.Rc.html
core/intrinsics/fn.drop_in_place.html:61: broken link - core/vec/struct.Vec.html
core/intrinsics/fn.drop_in_place.html:66: broken link - core/intrinsics/fn.read.html
std/intrinsics/fn.drop_in_place.html:59: broken link - std/intrinsics/fn.read.html
std/intrinsics/fn.drop_in_place.html:66: broken link - std/intrinsics/fn.read.html
```

**Comment by GuillaumeGomez** [comment]
Failed:

```bash
core/intrinsics/fn.drop_in_place.html:59: broken link - core/intrinsics/fn.read.html
core/intrinsics/fn.drop_in_place.html:61: broken link - core/boxed/struct.Box.html
core/intrinsics/fn.drop_in_place.html:61: broken link - core/rc/struct.Rc.html
core/intrinsics/fn.drop_in_place.html:61: broken link - core/vec/struct.Vec.html
core/intrinsics/fn.drop_in_place.html:66: broken link - core/intrinsics/fn.read.html
std/intrinsics/fn.drop_in_place.html:59: broken link - std/intrinsics/fn.read.html
std/intrinsics/fn.drop_in_place.html:66: broken link - std/intrinsics/fn.read.html
thread 'main' panicked at 'found some broken links', src/tools/linkchecker/main.rs:54
```

**Comment by steveklabnik** [comment]
@matthew-piziak ping! still keeping up with this?

**Comment by matthew-piziak** [comment]
@steveklabnik Sorry, I've been traveling for the past three weeks. I should be back in the game around 7 October. Right now this PR is soft-blocking on https://github.com/rust-lang/rust/issues/36417. Feel free to close; I'll keep the branch alive on my fork.

**Comment by steveklabnik** [comment]
Same as in the other thread, no worries, safe travels, let's close for now :smile:

**Pull request closed without merging**